templates/validation.hbs

```diff
@@ -303,5 +325,23 @@ export function timeToJson(name: string, value: string): string {
 	return timeFromJson(name, value)
 }
 
+export function verifyMinimumFiles(fieldName: string, minFileCount: number): (req: Request, res: Response, next: NextFunction) => void {
+	return function(req: Request, res: Response, next: NextFunction) {
+		if (!req.files || Array.isArray(req.files) || (req.files[fieldName] && req.files[fieldName].length < minFileCount)) {
+			res.status(400).send(`The field '${fieldName}' requires a minimum of ${minFileCount} file(s)`)
+		} else {
+			next()
+		}
+	}
+}
+
+export function fileFromValue(name: string, value: Express.Multer.File[] | Express.Multer.File): any {
```

Comment:
Does this ever get called with an array? If there is an array of files it should be an array in the schema, and we'd handle the array using `arrayFromJson`, which would ensure we're just called with a single file.

What happens if we remove the array from the type and just have this function just return `value`? Does everything still work?

Reply:
It gets called with an array only when the field supports a single file upload. For fields supporting multiple files, `arrayFromJson` is called on it so it ends up only taking file objects. When I removed the array type and returned value, the `request` object in the implementation ends up with the array containing the file object for single file uploads.

Reply:
For arrays of files, it gets called with the individual file objects because the top-level type of that schema is an array, so the generator knows to process it with `arrayFromJson` first.

Reply:
@Dakskihedron thanks for this reasoning; it isn't clear from the implementation why this is, so we should add a comment in the `isArray` block below to explain it... OR that might trigger us to ask, why are we putting a file array into our request when we only want a single file and then having to deal with it here. That's daft.

SO

Let's make this function only accept a single file and ditch the array stuff.

Let's modify `api.hbs` so we pull out `[0]` from the `req.files["blah"]` if it's not an array property. You can add an `isArray` property into `FileUploadProperty` as a convenience.

Reply:
We could probably remove the `fileFromValue` function since its purpose was to extract the file object from the array.